airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -210,7 +220,13 @@ else if (hasFailed.get()) {
           .withMaxSecondsBeforeSourceStateMessageEmitted(messageTracker.getMaxSecondsToReceiveSourceStateMessage())
           .withMeanSecondsBeforeSourceStateMessageEmitted(messageTracker.getMeanSecondsToReceiveSourceStateMessage())
           .withMaxSecondsBetweenStateMessageEmittedandCommitted(messageTracker.getMaxSecondsBetweenStateMessageEmittedAndCommitted().orElse(null))
-          .withMeanSecondsBetweenStateMessageEmittedandCommitted(messageTracker.getMeanSecondsBetweenStateMessageEmittedAndCommitted().orElse(null));
+          .withMeanSecondsBetweenStateMessageEmittedandCommitted(messageTracker.getMeanSecondsBetweenStateMessageEmittedAndCommitted().orElse(null))
+          .withReplicationStartTime(replicationStartTime)
+          .withReplicationEndTime(replicationEndTime > 0 ? replicationEndTime : null)
```

Comment:
and similarly for end time, I think we can rely on the existing metric for sync end time, which is added to the SyncSummary below (line 267: .withEndTime(System.currentTimeMillis()))

if we add it here too, it might end up being confusing since we'll have 2 slightly different end times for syncs.

Reply:
Oh - are you trying to get a more replication-specific timing that doesn't include all this extra stuff happening below? If so, maybe we could just change how we track the existing sync end times. It would mean there are inconsistencies in how the end time was being calculated previously and going forward, but it might be ok, depending on how this data is being used. 

I would still lean toward not having 2 different replication start & end times stored on the sync summary, but ultimately it is up to you!

Reply:
For timing information, I think we might want *all* of these times.   For example, say we are curious about the time it takes the median sync to "boot up" - which might be something the cloud team is interested in.  That's sourceReadStartTime - startTime.  

I opted not to change the definitions or names of our existing metrics, which happen to be the outer boundaries.  I'll update the image above to be more specific with the new labels.

cc @ChristopheDuong and @davinchia, who might be interested in these new times.